org.civicrm.reqangular/src/common/directives/horizontal-scroll-shadows.directive.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-env amd */
+/* globals MutationObserver */
+
+define([
+  'common/angular',
+  'common/modules/directives'
+], function (angular, directives) {
+  'use strict';
+
+  directives.directive('horizontalScrollShadows', [function () {
+    return {
+      restrict: 'A',
+      controller: horizontalScrollShadowsController
+    };
+  }]);
+
+  horizontalScrollShadowsController.$inject = ['$element', '$scope', '$window'];
+
+  function horizontalScrollShadowsController ($element, $scope, $window) {
+    var content, wrapper;
+
+    (function init () {
+      wrapContents();
+      initToggleShadowsHandlers();
+      toggleShadows();
+    }());
+
+    /**
+     * Makes shadow toggle on the wrapper element scrolling,
+     * or DOM changes inside content element, or window resize
+     */
+    function initToggleShadowsHandlers () {
+      wrapper.on('scroll', toggleShadows);
+      angular.element($window).on('resize', toggleShadows);
+      (new MutationObserver(toggleShadows)).observe(content[0], { subtree: true, childList: true });
+      $scope.$watch(function () {
+        return wrapper[0].clientWidth;
+      }, toggleShadows);
+      $scope.$watch(function () {
+        return content[0].clientWidth;
```

Comment:
why are you watching `clientWidth` twice? you meant `clientHeight` ?

Reply:
why do you use both the mutation observer and `$watch` ?

Reply:
I think I attempted to describe that in the PR, but let me clarify a bit more.

`MutationObserver` tracks DOM manipulation, for example, when the content is being loaded. When DOM is populated, that will **not** change the `clientWidth`.

`ClientWidth`, on the other hand,  is needed when you dynamically change the style of the content **or** wrapper. DOM is not touched in this case, so `MutationObserver` will **not** be  triggered.